api/coreservice.go

```diff
@@ -1418,9 +1418,34 @@ func (core *coreService) correctLogsRange(start, end uint64) (uint64, uint64, er
 	return start, end, nil
 }
 
-// CalculateGasConsumption estimate gas consumption for actions except execution
-func (core *coreService) CalculateGasConsumption(intrinsicGas, payloadGas, payloadSize uint64) (uint64, error) {
-	return action.CalculateIntrinsicGas(intrinsicGas, payloadGas, payloadSize)
+// EstimateGasForNonExecution estimates action gas except execution
+func (core *coreService) EstimateGasForNonExecution(actType action.Action, payload []byte) (uint64, error) {
+	var intrinsicGas, payloadGas, payloadSize uint64
+	switch act := actType.(type) {
+	case *action.Transfer:
+		intrinsicGas, payloadGas, payloadSize = action.TransferBaseIntrinsicGas, action.TransferPayloadGas, uint64(len(payload))
```

Comment:
it is a bit weird to have gas estimation calculation outside of action.Action and hardcoded in core service instead in this way.

Reply:
Should `EstimateGasForNonExecution()` be moved to `action.go`